weld/llvm.rs

```diff
@@ -901,7 +903,22 @@ impl LlvmGenerator {
                     self.prelude_code.add(&name_replaced);
                     self.prelude_code.add("\n");
 
-                    // Supports vectorization, so splice in the vector extensions.
+                    // Generate comparison functions, using a special template for vec[i8].
+                    if *elem.as_ref() == Scalar(ScalarKind::I8) {
```

Comment:
Ah, because it's overriding the lexicographic comparison function. It might make more sense to add an explicit equality function for `==`, so we can just do a byte-level comparison for any scalar-based vector.